community/kernel/src/main/java/org/neo4j/kernel/impl/api/KernelTransactionImplementation.java

```diff
@@ -144,14 +144,14 @@ TransactionWriteState upgradeToSchemaWrites() throws InvalidTransactionTypeKerne
     private final StorageStatement storageStatement;
     private CloseListener closeListener;
     private AccessMode accessMode;
-    private Locks.Client locks;
+    private volatile Locks.Client locks;
     private boolean beforeHookInvoked;
-    private boolean closing, closed;
+    private volatile boolean closing, closed;
     private boolean failure, success;
     private volatile Status terminationReason;
     private long startTimeMillis;
     private long lastTransactionIdWhenStarted;
-    private long lastTransactionTimestampWhenStarted;
```

Comment:
That's a couple more fields made `volatile`, could we piggy-back some access on some other? Think about it, if we can't, we can't